packages/server/src/integrations/couchdb.ts

```diff
@@ -79,21 +82,19 @@ class CouchDBIntegration implements IntegrationBase {
       connected: false,
     }
     try {
-      const result = await this.query("exists", "validation error", {})
-      response.connected = result === true
+      response.connected = await this.query(
+        () => this.client.exists(),
+        "validation error"
+      )
     } catch (e: any) {
       response.error = e.message as string
     }
     return response
   }
 
-  async query(
-    command: string,
-    errorMsg: string,
-    query: { json?: object; id?: string }
-  ) {
+  async query<T>(operation: () => Promise<T>, errorMsg: string) {
```

Comment:
This function isn't really doing what it advertises. It runs a given function and logs the error before rethrowing. It could be better called `logError(...` or gotten rid of entirely, I don't personally look at the error logs when those same errors are more usefully visible in traces.

Reply:
Fair point - simplified!